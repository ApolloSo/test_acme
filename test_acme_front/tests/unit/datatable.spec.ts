import { createLocalVue, mount } from '@vue/test-utils';
import AcmeTable from '@/components/AcmeTable.vue';

import Vue from 'vue';
import Vuetify from 'vuetify';
Vue.use(Vuetify);

describe('AcmeTable.vue', () => {
  const localVue = createLocalVue();
  const vuetify = new Vuetify();
  // beforeEach(() => {
  //   vuetify: new Vuetify()
  // })

  it('renders props.data when passed', () => {
    const data = [{id: 'test_id', description: 'test description'}];
    const wrapper = mount(AcmeTable, {
      localVue,
      vuetify,
      propsData: { data },
    });
    // expect(wrapper.find('test').exists()).toBe(true)
    // expect(wrapper.html()).toMatchSnapshot();
    const cell = wrapper.find('tbody > tr > td');
    expect(cell.text()).toBe('test_id');
  });
});
