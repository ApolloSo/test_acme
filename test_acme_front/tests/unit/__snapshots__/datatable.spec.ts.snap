// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcmeTable.vue renders props.data when passed 1`] = `
<div class="container container--fluid">
  <div class="v-data-table v-data-table--has-bottom theme--light">
    <div class="v-data-table__wrapper">
      <table>
        <colgroup>
          <col class="">
          <col class="">
          <col class="">
          <col class="">
          <col class="">
          <col class="">
          <col class="">
          <col class="">
          <col class="">
        </colgroup>
        <thead class="v-data-table-header">
          <tr>
            <th role="columnheader" scope="col" aria-label="ID: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>ID</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="Name" class="text-start" style="width: 200px; min-width: 200px;"><span>Name</span></th>
            <th role="columnheader" scope="col" aria-label="Code: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Code</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="Opening date: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Opening date</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="Store type: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Store type</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="Season: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Season</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="Special f1: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Special f1</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="Special f2: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Special f2</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="Description: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Description</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
          </tr>
        </thead>
        <tbody>
          <tr class="">
            <td class="text-start">test_id</td>
            <td class="text-start">
              <div class="d-flex"><span></span>
                <!----> <button type="button" class="v-icon notranslate v-icon--link mdi mdi-lead-pencil theme--light" style="font-size: 16px;"></button>
              </div>
            </td>
            <td class="text-start"></td>
            <td class="text-start"></td>
            <td class="text-start"></td>
            <td class="text-start"></td>
            <td class="text-start"></td>
            <td class="text-start"></td>
            <td class="text-start">

            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div class="v-data-footer">
      <div class="v-data-footer__select">Rows per page:<div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
          <div class="v-input__control">
            <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-16" class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <div class="v-select__selection v-select__selection--comma">10</div><input aria-label="Rows per page:" id="input-16" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden" value="10">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-data-footer__pagination">1-1 of 1</div>
      <div class="v-data-footer__icons-before"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Previous page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></span></button></div>
      <div class="v-data-footer__icons-after"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Next page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></span></button></div>
    </div>
  </div>
</div>
`;
